Library/Homebrew/os/mac/hardware.rb

```diff
@@ -79,7 +79,11 @@ def cores
   end
 
   def bits
-    sysctl_bool("hw.cpu64bit_capable") ? 64 : 32
+    # hw.cpu64bit_capable is not available on Tiger, but it is
+    # possible to detect 64-bit CPUs by checking the ppc- and
+    # intel-specific flags, as shown in
+    # https://opensource.apple.com/source/xnu/xnu-1504.7.4/tools/tests/xnu_quick_test/misc.c?txt
+    sysctl_bool("hw.optional.64bitops") || sysctl_bool("hw.optional.x86_64") ? 64 : 32
```

Comment:
Yes, `hw.optional.x86_64` is available on all recent versions (of course `hw.optional.64bitops` will be absent, as it is only available when the architecture is ppc). It is also kind of documented in the [`sys/sysctl.h` header](https://github.com/phracker/MacOSX-SDKs/blob/master/MacOSX10.11.sdk/usr/include/sys/sysctl.h#L669-L687).